### ISSUE TITLE
Cannot crop on iOS 11

### ISSUE THREAD

**Issue by j-down** [ytakzk/Fusuma#215]
The image cropping in the camera roll view does not work for iOS 11. Please fix as soon as possible, thank you for your time.

**Comment by francodriansetti** [comment]
For me work properly, can u give us an example please?

**Comment by j-down** [comment]
Of course! For my product, Clique, we use it to crop images posted to the
app. Here is the process. Instead of cropping as a square, it uses the
entire photo from the camera roll.

The screenshots are attached. I hope this helps. Thank you so much for your
time!

**Comment by j-down** [comment]
![img_1905](https://user-images.githubusercontent.com/14189826/30956427-449e20b0-a405-11e7-9148-fc5fd158169b.PNG)
![img_1904](https://user-images.githubusercontent.com/14189826/30956425-449bc572-a405-11e7-8d33-c5032bcd3455.PNG)
![img_1903](https://user-images.githubusercontent.com/14189826/30956426-449d084c-a405-11e7-8501-e9d5677176a7.PNG)

**Comment by francodriansetti** [comment]
@j-down thanks for screenshot. 

Did you tried to set: cropHeightRatio = 1 // which means a squared-size photo

**Comment by j-down** [comment]
Before it seemed to auto crop without setting that (I assume 1 is the default?). However, I just tried setting it to 1 manually and it still does not crop the image.

**Comment by francodriansetti** [comment]
Try with demo app, I'm not able to replicate the issue

**Comment by j-down** [comment]
![img_1907](https://user-images.githubusercontent.com/14189826/30958033-e5610d1a-a409-11e7-8d0d-a2c88cf767f1.PNG)
![img_1906](https://user-images.githubusercontent.com/14189826/30958034-e57638ac-a409-11e7-81e0-27e3d4e864d4.PNG)

**Comment by j-down** [comment]
Just tried with demo app - same error. :/ are you testing this with iOS 11?

**Comment by francodriansetti** [comment]
Yes iOS 11, branch Swift-4

**Comment by JerryAndrews** [comment]
From my testings on iPhone 7 Plus running iOS 11:

When selecting a vertical or horizontal screenshot, that has not been modified, fusuma does not crop the image. It returns the full horizontal or vertical screenshot.

It appears @j-down had a similar issue when selecting screenshots specifically. 

It appears to me that the issue is when the width/height of the image is equivalent to the device width/height fusuma is not able to crop. Hence why it does not crop with screenshots.

To test this theory:

I took one of the horizontal screenshots from my device, and cropped a minute portion of the width. When I proceeded to crop in fusuma I got the desired image.

I took one of the vertical screenshots from my device, and cropped a minute portion of the height. When I proceeded to crop in fusuma I got the desired image.

I tested with an image of dimensions 5500 × 3637. I was able to crop without issue. 

I tested with a picture taken from the front facing cam.  I was able to crop without issue. 

I tested with a picture taken from the rear facing cam.  I was able to crop without issue. 

In summary the issue is only happening with device screenshots.

**Comment by shrawan2015** [comment]
Thanks for providing great library.

Even I could not crop the images from gallery and  photos in iOS 11. Is there any properties that need to be set to enable the crop in the library.
